### PR TITLE
PLANET-3405 add new page

### PR DESCRIPTION
**Pull request by angtheod**
First implementation of the EN Form Add/Edit page. Can save/load dummy fields. Re-arranged the sub-menu items to resemble WP's Post/Page sub-menu items order.

### CODE REVIEW

File: classes/controller/menu/class-enform-post-controller.php [hunk 1]
```diff
@@ -140,6 +143,129 @@ public function modify_post_row_actions( $actions, $post ) {
...
+				[
+					'id'           => $prefix . 'metabox',
+					'title'        => __( 'Fields', 'planet4-engagingnetworks' ),
+					'object_types' => [ self::POST_TYPE ],
+				]
+			);
+
+			$meta_box->add_field(
+				[
+					'name' => __( 'Available Fields', 'planet4-engagingnetworks' ),
```
Comment (sagarsdeshmukh) [inline classes/controller/menu/class-enform-post-controller.php]: Array double arrow not aligned correctly.(PHPCS)

File: classes/controller/menu/class-enform-post-controller.php [hunk 2]
```diff
@@ -140,6 +143,129 @@ public function modify_post_row_actions( $actions, $post ) {
...
+						'value'    => 'example@example.com',
+						'required' => true,
+						'hidden'   => false,
+					],
+				],
+				'domain'     => 'planet4-engagingnetworks',
+			] );
+		}
+
+		/**
```
Comment (sagarsdeshmukh) [inline classes/controller/menu/class-enform-post-controller.php]: Please add doc comment .

File: classes/controller/menu/class-enform-post-controller.php [hunk 3]
```diff
@@ -140,6 +143,129 @@ public function modify_post_row_actions( $actions, $post ) {
...
+				[
+					'id'           => $prefix . 'metabox',
+					'title'        => __( 'Questions', 'planet4-engagingnetworks' ),
+					'object_types' => [ self::POST_TYPE ],
+				]
+			);
+
+			$meta_box->add_field(
+				[
+					'name' => __( 'Available Questions', 'planet4-engagingnetworks' ),
```
Comment (sagarsdeshmukh) [inline classes/controller/menu/class-enform-post-controller.php]: Same here the array double arrow alignment.

File: classes/controller/menu/class-enform-post-controller.php [hunk 4]
```diff
@@ -140,6 +143,129 @@ public function modify_post_row_actions( $actions, $post ) {
...
+					'options' => [
+						'check1' => 'Check One',
+						'check2' => 'Check Two',
+						'check3' => 'Check Three',
+					],
+				]
+			);
+		}
+
+		/**
```
Comment (sagarsdeshmukh) [inline classes/controller/menu/class-enform-post-controller.php]: Please add doc comment.

File: classes/controller/menu/class-enform-post-controller.php [hunk 5]
```diff
@@ -140,6 +143,129 @@ public function modify_post_row_actions( $actions, $post ) {
...
+				__( 'Selected Components', 'planet4-engagingnetworks' ),
+				[ $this, 'view_meta_box_selected' ],
+				[ self::POST_TYPE ],
+				'normal',
+				'high',
+				$post
+			);
+		}
+
+		/**
```
Comment (sagarsdeshmukh) [inline classes/controller/menu/class-enform-post-controller.php]: Please add short desc in Doc comment

File: classes/controller/menu/class-enform-post-controller.php [hunk 6]
```diff
@@ -140,6 +143,129 @@ public function modify_post_row_actions( $actions, $post ) {
...
+		 * @param \WP_Post $post The currently Added/Edited EN Form.
+		 */
+		public function view_meta_box_selected( $post ) {
+			$this->view->selected_meta_box( [
```
Comment (sagarsdeshmukh) [inline classes/controller/menu/class-enform-post-controller.php]: PHPCS - please move array opening tag on new line.

File: classes/view/class-view.php [hunk 7]
```diff
@@ -121,6 +121,15 @@ public function block( $template_name, $data, $template_ext = 'twig', $relevant_
 			}
 		}
 
+		/**
+		 * Render the settings page of the plugin.
```
Comment (sagarsdeshmukh) [inline classes/view/class-view.php]: Doc shot desc need to change.

File: includes/selected_meta_box.twig [hunk 8]
```diff
@@ -0,0 +1,46 @@
+{% block selected_meta_box %}
+	<table id="en_form_selected_fields_table" class="table table-striped table-bordered" cellspacing="0" width="100%">
+		<thead>
+		<tr valign="top">
+			<th>{{ __( 'Name', domain ) }}</th>
```
Comment (kirdia) [inline includes/selected_meta_box.twig]: please use the actual name for the textdomain and not a variable

File: classes/controller/menu/class-enform-post-controller.php [hunk 9]
```diff
@@ -88,7 +99,7 @@ public function register_post_type() {
 					'rewrite'             => false,
 					'query_var'           => false,
 					'public'              => false,
-					'publicly_queryable'  => false,
+					'publicly_queryable'  => true,
```
Comment (kirdia) [inline classes/controller/menu/class-enform-post-controller.php]: @angtheod is this necessary?

Comment (angtheod) [reply classes/controller/menu/class-enform-post-controller.php]: oh, nop, forgot to switch it back after checking what it does

**Pull request merged**